pkgs/by-name/q2/q2pro/package.nix

```diff
@@ -0,0 +1,87 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, meson
+, pkg-config
+, ninja
+, zlib
+, libpng
+, libjpeg
+, curl
+, SDL2
+, openal
+, openalSoft
+, libogg
+, libvorbis
+, libXi
+, wayland
+, wayland-protocols
+, libdecor
+, ffmpeg
+, x11Support ? stdenv.isLinux
+, waylandSupport ? stdenv.isLinux
+}:
+
+stdenv.mkDerivation rec {
+  pname = "q2pro";
+  version = "0-unstable-2023-12-16";
+
+  src = fetchFromGitHub {
+    owner = "skullernet";
+    repo = pname;
+    rev = "d7e4d98b51a69e4bc0f9394626d7227e59c4c628";
+    hash = "sha256-2xxdpz4PrOk1YjdrNY98Ypsektvl2v5zINWZA7XYo3g=";
+  };
+
+  nativeBuildInputs = [
+    meson
+    pkg-config
+    ninja
+  ];
+
+  buildInputs = [
+    zlib
+    libpng
+    libjpeg
+    curl
+    SDL2
+    libogg
+    libvorbis
+    ffmpeg
+  ] ++ lib.optionals stdenv.isLinux [
+    openal
+  ] ++ lib.optionals stdenv.isDarwin [
+    openalSoft
```

Comment:
[q2pro relies on <(Open)AL/atext.h>](https://github.com/skullernet/q2pro/blob/master/src/client/sound/qal.h#L29) which is not present in OpenAL.framework on macOS, therefore opting to use [openal-soft](https://github.com/kcat/openal-soft) instead.

Reply:
no longer relevant as the author [recommends using openal-soft for all cases]:(https://github.com/skullernet/q2pro/blob/r3510/INSTALL.md)
> OpenAL sound backend requires OpenAL Soft development headers for compilation. At runtime, OpenAL library from any vendor can be used (but OpenAL Soft is strongly recommended).